go/kbfs/simplefs/simplefs.go

```diff
@@ -3022,3 +3010,55 @@ func (k *SimpleFS) SimpleFSConfigureDownload(
 	k.downloadManager.configureDownload(arg.CacheDirOverride, arg.DownloadDirOverride)
 	return nil
 }
+
+// SimpleFSGetGUIFileContext implements the SimpleFSInterface.
+func (k *SimpleFS) SimpleFSGetGUIFileContext(ctx context.Context,
+	path keybase1.KBFSPath) (resource keybase1.GUIFileContext, err error) {
+	if len(path.Path) == 0 {
+		return keybase1.GUIFileContext{}, errors.New("empty path")
+	}
+
+	if k.localHTTPServer == nil {
+		return keybase1.GUIFileContext{}, errors.New("HTTP server is disabled")
+	}
+	// Refresh the token every time. This RPC is called everytime a file is
+	// being viewed and we have a cache size of 64 so this shouldn't be a
+	// problem.
+	token, err := k.localHTTPServer.NewToken()
+	if err != nil {
+		return keybase1.GUIFileContext{}, err
+	}
+	address, err := k.localHTTPServer.Address()
+	if err != nil {
+		return keybase1.GUIFileContext{}, err
+	}
+
+	var encodedSegments []string
+	for _, segment := range strings.Split(path.Path, "/") {
+		encodedSegments = append(encodedSegments, url.PathEscape(segment))
+	}
+
+	u := url.URL{
+		Scheme:   "http",
+		Host:     address,
+		Path:     "/files" + strings.Join(encodedSegments, "/"),
+		RawQuery: "token=" + token,
+	}
+
+	r, err := http.NewRequest("HEAD", u.String(), nil)
+	if err != nil {
+		return keybase1.GUIFileContext{}, err
+	}
+	resp, err := k.httpClient.Do(r)
```

Comment:
Going through the HTTP client here seems pretty heavyweight.  Is there no easy way to fake a header without doing that?

Reply:
It's what we've been doing on the GUI side, and this shouldn't be worse that. I think we could avoid the http part by duplicating [this](https://github.com/golang/go/blob/8b6fa668ee1aaf6c00073115cc09957195ee30e1/src/net/http/fs.go#L196-L201), but if the logic there ever changes we'd get a wrong value here. Though with the invariance thing, it seems at least we won't end up serving wrong content-type (which could lead to security concerns). After typing this out it does seem like a good idea, so I'll be going with this route and update the PR.